src/segment-loader.js

```diff
@@ -364,8 +369,8 @@ export default class SegmentLoader extends videojs.EventTarget {
     if (currentBuffered.length === 0) {
       // find the segment containing currentTime
       mediaIndex = getMediaIndexForTime(playlist,
-                                        currentTime,
-                                        this.expired_ + this.timeCorrection_);
+                                        currentTime + this.timeCorrection_,
+                                        this.expired_);
```

Comment:
The idea here is that expired will only ever be used when we do *not* have accurate timing data in the manifest. In most cases we do so the original code was not advancing the fetcher. Now we are directly fudging the `currentTime` so we force the fetcher to walk forward even if we have some (possibly incorrect) information recorded into the segment annotation.